src/rust/engine/process_execution/src/remote.rs

```diff
@@ -105,8 +108,8 @@ impl CommandRunner {
           // TODO: Add a timeout of some kind.
           // https://github.com/pantsbuild/pants/issues/5504
 
-          .and_then(move |(execute_request, operation)| {
-            future::loop_fn(operation, move |operation| {
+          .and_then(move |(execute_request, operation, iter_num)| {
```

Comment:
Nit: No need to pass iter_num across the futures, can just use 0 in the tuple in the first arg to loop_fn